cmd/kubeadm/app/cmd/upgrade/plan.go

```diff
@@ -52,21 +61,182 @@ func NewCmdPlan(apf *applyPlanFlags) *cobra.Command {
 				return err
 			}
 
-			return runPlan(flags, userVersion)
+			printer, err := outputFlags.ToPrinter()
+			if err != nil {
+				return err
+			}
+
+			return runPlan(flags, userVersion, printer)
 		},
 	}
 
+	outputFlags.AddFlags(cmd)
+
 	// Register the common flags for apply and plan
 	addApplyPlanFlags(cmd.Flags(), flags.applyPlanFlags)
 	return cmd
 }
 
+// newComponentUpgradePlan helper creates outputapiv1alpha1.ComponentUpgradePlan object
+func newComponentUpgradePlan(name, currentVersion, newVersion string) *outputapiv1alpha1.ComponentUpgradePlan {
+	return &outputapiv1alpha1.ComponentUpgradePlan{
+		Name:           name,
+		CurrentVersion: currentVersion,
+		NewVersion:     newVersion,
+	}
+}
+
+// upgradePlanPrintFlags defines printer flag structure for the
+// upgrade plan kubeadm command and provides a method
+// of retrieving a known printer based on flag values provided.
+type upgradePlanPrintFlags struct {
```

Comment:
not a blocking comment, but 
it feels to me PrintFlags should have been an interface that different outputs implement.
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/output/output.go#L41

- e.g. upgradePlanPrintFlags being an implementation of PrintFlags.

re-defining a similar structure for the different outputs doesn't seem right.

Reply:
I'm not sure I understand how defining an interface helps here. It's a local structure used only to define couple of methods. Even if I group those methods into the interface how I suppose to use that?

Reply:
it is an OOP problem that can be solved.

https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/output/output.go#L38-L40
with this comment we claim that this structure will be reused, but it's not used in this file.

Reply:
Yes, it will be reused at least in 'config images list' output. It can't be reused in the upgrade plan ouput due to buffered type of output that you've requested.